src/useData.test.tsx

```diff
@@ -0,0 +1,20 @@
+import { renderHook } from '@testing-library/react';
+
+import {
+    QueryType,
+    IDataQuery,
+    useData
+} from './useData';
+
+describe('useData', () => {
+    test('returns data for custom date range', () => {
+        const query: IDataQuery = {
+            queryString: '01/03/2022-03/03/2022',
+            queryType: QueryType.RANGE
+        };
+
+        const { result } = renderHook(() => useData(query));
```

Comment:
Now thinking of it similary to using fetch or axios whatever async code the custom hook uses underneath could be stubbed to deal with sync code only - that would leave just setState as async piece.

Reply:
...also no `waitForTheNextUpdate` is available in `renderHookResult` anymore.

Reply:
I wish I found it sooner…
https://stackoverflow.com/questions/73150220/cannot-test-custom-hooks-with-react-18-and-renderhook-from-testing-library-react